### PR TITLE
[core][autoscaler] Include failed request info from autoscaler

### PR DESCRIPTION
**Pull request by rickyyx**
## Why are these changes needed?
This allows us to show failure launch request info from autoscaler. 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by rickyyx** [comment]
@hanming-lu

File: src/ray/protobuf/experimental/autoscaler.proto [hunk 1]
```diff
@@ -190,6 +190,24 @@ message PendingInstanceRequest {
...
+message FailedInstanceRequest {
+  // The instance type name for the cloud, e.g. t1.micro
+  string instance_type_name = 1;
+  // The node type name, e.g. ray-head-node, matching `available_node_types`
+  // in the autoscaler config. See `ray/autoscaler/ray-schema.json`
+  string ray_node_type_name = 2;
+  // The number of such instance requested to be started.
+  int32 count = 3;
+  // Descriptive reasons for the failure.
+  string reason = 4;
```
Comment (hanming-lu) [inline src/ray/protobuf/experimental/autoscaler.proto:206]: fyi, for reason, the msg can be long in size.

**Review by hanming-lu — APPROVED** [review]

File: src/ray/protobuf/experimental/autoscaler.proto (hunk 1 above)
Comment (rickyyx) [reply src/ray/protobuf/experimental/autoscaler.proto:206]: This is fine?

Comment (hanming-lu) [reply src/ray/protobuf/experimental/autoscaler.proto:206]: yup, just fyi

**Review by brucez-anyscale — APPROVED** [review]

**Review by scv119 — APPROVED** [review]

**Pull request merged**